### PR TITLE
Minor code fixes

### PR DESCRIPTION
**Pull request by matriv**
- Unused args, methods, classes.
- Use newHashSet/Map calls
- Minor code formatting issues
- Fix compiler warnings

### CODE REVIEW

File: server/src/main/java/io/crate/analyze/where/WhereClauseAnalyzer.java [hunk 1]
```diff
@@ -143,20 +143,21 @@ public static PartitionResult resolvePartitions(Symbol query,
...
         }
 
+        if (queryPartitionMap.isEmpty()) {
+            return new PartitionResult(Literal.BOOLEAN_FALSE, Collections.emptyList());
+        }
         if (queryPartitionMap.size() == 1) {
             Map.Entry<Symbol, List<Literal<?>>> entry = Iterables.getOnlyElement(queryPartitionMap.entrySet());
             return new PartitionResult(
                 entry.getKey(), Lists.map(entry.getValue(), literal -> nullOrString(literal.value())));
-        } else if (queryPartitionMap.size() > 0) {
```
Comment (jeeminso) [inline server/src/main/java/io/crate/analyze/where/WhereClauseAnalyzer.java:150]: I think this changes the execution path taken for `queryPartitionMap.size() == 0`

**Review by jeeminso — APPROVED** [review]
Thank you! Had one minor comment.

File: server/src/main/java/io/crate/analyze/where/WhereClauseAnalyzer.java (hunk 1 above)
Comment (matriv) [reply server/src/main/java/io/crate/analyze/where/WhereClauseAnalyzer.java:150]: if it's empty, is moved first, then if == 1 and else if it's > 1, do you still see an issue, should I revert?

Comment (jeeminso) [reply server/src/main/java/io/crate/analyze/where/WhereClauseAnalyzer.java:150]: Oh sorry didn't see that. Please proceed.

**Pull request merged**